### PR TITLE
Improve the Ubuntu Server linkbot monitoring

### PR DESCRIPTION
**Pull request by oleorhagen**
Now takes the new image link explicitly from the sub `/release` domain, through matching on link names, as opposed to generating them through a regular expression, like was previously done.

### CODE REVIEW

**Comment by oleorhagen** [comment]
@mzedel None urgent

File: scripts/linkbot/monitor-ub-server.js [hunk 1]
```diff
@@ -33,14 +34,28 @@ JSDOM.fromURL(url, {}).then(dom => {
...
-        } else {
-            newLine = `UBUNTU_SERVER_RPI_IMAGE_URL=\"${url}${matches[0].groups.release}.04/release/ubuntu-${matches[0].groups.release}.04-preinstalled-server-armhf+raspi.img.xz\"`
+
+    return matchOn;
+
+    // Get the release image url from the releases (sub)-page
+    // const url = "http://cdimage.ubuntu.com/ubuntu/releases/"
+}).then(releaseVersion => {
+    var releaseVersion = releaseVersion.replace(/\s/g, "").replace(/\//g, "")
+    JSDOM.fromURL(url + releaseVersion + "/release/", {}).then(dom => {
```
Comment (mzedel) [inline scripts/linkbot/monitor-ub-server.js:44]:
```suggestion
    JSDOM.fromURL(`${url}${releaseVersion}/release/`, {}).then(dom => {
```
just for consistency 😉

**Review by mzedel — APPROVED** [review]
👍 I sense a ~darkness~ Javascript in you!
 and again: feel free to ignore the suggestions...

File: scripts/linkbot/monitor-ub-server.js [hunk 2]
```diff
@@ -33,14 +34,28 @@ JSDOM.fromURL(url, {}).then(dom => {
...
+            })
+            .reduce((acc, ref) => {
+                acc.push(ref.href)
+                return acc
+            }, [])
+
+        if (matches.length > 0) {
+            console.log(`Ubuntu server image has a new release: ${matches[0]}`)
+            updateURLLink(`${target}=${matches[0]}`, target)
         }
```
[the remark this answers is not in the document]
Comment (oleorhagen) [reply scripts/linkbot/monitor-ub-server.js]: Beautiful! :heart:

**Pull request merged**